libavcodec/libdavs2.c

```diff
@@ -0,0 +1,206 @@
+        }
+        memcpy(frame->data[i], pic->planes[i], size_plane);
+    }
+
+    frame->width     = cad->headerset.horizontal_size;
+    frame->height    = cad->headerset.vertical_size;
+    frame->pts       = cad->out_frame.pts;
+    frame->pict_type = pic->type;
+    frame->format    = avctx->pix_fmt;
+
+    cad->decoded_frames++;
+    return 1;
+}
+
+static av_cold ff_davs2_end(AVCodecContext *avctx)
+{
+    DAVS2Context *cad = avctx->priv_data;
+
+    /* close the decoder */
+    if (cad->decoder) {
+        davs2_decoder_close(cad->decoder);
+        av_log(avctx, AV_LOG_VERBOSE, "decoder destroyed. 0x%p; frames %d\n", cad->decoder, cad->decoded_frames);
+        cad->decoder = NULL;
+    }
+
+    return 0;
+}
+
+static int davs2_decode_frame(AVCodecContext *avctx, void *data, int *got_frame, AVPacket *avpkt)
+{
+    DAVS2Context *cad = avctx->priv_data;
+    int buf_size       = avpkt->size;
+    uint8_t *buf_ptr = avpkt->data;
+    AVFrame *frame = data;
+    int ret = 0;
+
+    *got_frame = 0;
+    cad->frame = frame;
+    avctx->flags |= AV_CODEC_FLAG_TRUNCATED;
+
+    if (!buf_size) {
+        cad->packet.data = buf_ptr;
+        cad->packet.len  = buf_size;
+        cad->packet.pts  = avpkt->pts;
+        cad->packet.dts  = avpkt->dts;
+
+        for (;;) {
+            ret = davs2_decoder_flush(cad->decoder, &cad->headerset, &cad->out_frame);
+
+            if (ret < 0)
+                return 0;
+
+            if (cad->out_frame.ret_type != DAVS2_DEFAULT) {
+                *got_frame = DumpFrames(avctx, &cad->out_frame, &cad->headerset, frame);
+                davs2_decoder_frame_unref(cad->decoder, &cad->out_frame);
+            }
+            if (*got_frame == 1)
+                break;
+        }
+        return 0;
+    } else {
+        for (; buf_size > 0;) {
+            int len = buf_size;   // for API-3, pass all data in
+
+            cad->packet.marker = 0;
+            cad->packet.data = buf_ptr;
+            cad->packet.len  = len;
+            cad->packet.pts  = avpkt->pts;
+            cad->packet.dts  = avpkt->dts;
+
+            len = davs2_decoder_decode(cad->decoder, &cad->packet, &cad->headerset, &cad->out_frame);
+
+            if (cad->out_frame.ret_type != DAVS2_DEFAULT) {
+                *got_frame = DumpFrames(avctx, &cad->out_frame, &cad->headerset, frame);
+                davs2_decoder_frame_unref(cad->decoder, &cad->out_frame);
+            }
+
+            if (len < 0) {
+                av_log(avctx, AV_LOG_ERROR, "A decoder error counted\n");
+                if (cad->decoder) {
+                    davs2_decoder_close(cad->decoder);
+                    av_log(avctx, AV_LOG_VERBOSE, "decoder destroyed. 0x%p; frames %d\n", cad->decoder, cad->decoded_frames);
+                    cad->decoder = NULL;
+                }
+                return AVERROR(EINVAL);
+            }
+
+            buf_ptr     += len;
+            buf_size    -= len;
+
+            if (*got_frame == 1)
```

Comment:
Is this a bool? just *got_frame or !*got_frame?  it the DumpFrames is a bool type?
and the API name "DumpFrames" is not ffmpeg code style , what about use davs_dump_frames?